src/core/tsi/alts/handshaker/alts_handshaker_client.cc

```diff
@@ -661,11 +661,22 @@ static void handshaker_client_destruct(alts_handshaker_client* c) {
     // TODO(apolcyn): we could remove this indirection and call
     // grpc_call_unref inline if there was an internal variant of
     // grpc_call_unref that didn't need to flush an ExecCtx.
-    grpc_core::ExecCtx::Run(
-        DEBUG_LOCATION,
-        GRPC_CLOSURE_CREATE(handshaker_call_unref, client->call,
-                            grpc_schedule_on_exec_ctx),
-        GRPC_ERROR_NONE);
+    if (grpc_core::ExecCtx::Get() == nullptr) {
+      // Unref handshaker call in a new exec_ctx if it has not been created.
+      grpc_core::ExecCtx exec_ctx;
```

Comment:
actually, can we simplify this to just inline `grpc_call_unref(call);`, in the case that there's no active ExecCtx? (because we're not running on a C-core thread in this case, the risk of holding arbitrary C-core locks down the call stack isn't there).

Reply:
Done.